### PR TITLE
Move rest of non-specd attributes to experimental

### LINKED ISSUE #1874: Hide all span attributes not defined by spec behind an experimental flag
Otherwise, if/when they are defined by the spec, we will have to keep the old ones around forever too if we don't want to introduce a breaking change.

### PR DESCRIPTION
**Pull request by trask**
Almost completes #1874, except for aws-sdk-2.2, because we don't have solution for `Config.get()` in library instrumentation yet.

### CODE REVIEW

**Review by iNikem — APPROVED** [review]

**Review by anuraaga — APPROVED** [review]

File: instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/decorators/KafkaSpanDecorator.java [hunk 1]
```diff
@@ -38,6 +39,11 @@
   private static final String TOPIC = "kafka.TOPIC";
   private static final String OFFSET = "kafka.OFFSET";
 
+  private static final boolean CAPTURE_EXPERIMENTAL_SPAN_ATTRIBUTES =
+      Config.get()
+          .getBooleanProperty(
+              "otel.instrumentation.apache-camel.experimental-span-attributes", false);
```
Comment (mateuszrzeszutek) [inline instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/decorators/KafkaSpanDecorator.java:45]: Hm, can you use the constant from `BaseSpanDecorator` if you make it protected?

**Review by mateuszrzeszutek — APPROVED** [review]

**Comment by mateuszrzeszutek** [comment]
@trask what's the problem with calling `Config.get()` in library instrumentations?

**Comment by trask** [comment]
> @trask what's the problem with calling `Config.get()` in library instrumentations?

`Config.get()` will return the default (empty) config for library instrumentation users

A couple possible solutions would be to init the default config with system properties for library instrumentation users, or to have an instrumentation-specific configuration API that library users (and the javaagent) can use to configure it.

File: instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/decorators/KafkaSpanDecorator.java (hunk 1 above)
Comment (trask) [reply instrumentation/apache-camel-2.20/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/apachecamel/decorators/KafkaSpanDecorator.java:45]: 👍 will make this change (in this PR) tomorrow

**Pull request merged**